src/Microsoft.ML.Data/Transforms/ColumnBindingsBase.cs

```diff
@@ -14,16 +14,40 @@
 
 namespace Microsoft.ML.Data
 {
-    internal abstract class SourceNameColumnBase
+    public class OneToOneColumn
```

Comment:
>OneToOneColumn [](start = 17, length = 14)

Do you mean this class to be used and directly instantiable? I would really not like that. I would prefer that our API be over sealed classes -- otherwise we'll have weird things like it being possible to assign `OneToOneColumn` taking APIs.

As a general policy, we prefer all classes to be either abstract or sealed. We have [generally followed this advice](https://blogs.msdn.microsoft.com/ericlippert/2004/01/22/why-are-so-many-of-the-framework-classes-sealed/).

Reply:
I am making it abstract, thanks for pointing it out.

---
In reply to: [267543430](https://github.com/dotnet/machinelearning/pull/3033#discussion_r267543430) [](ancestors = 267543430)